### PR TITLE
Add include and exclude filters for youtube feed

### PR DESCRIPTION
**Pull request by romangr**
Resolves https://github.com/umputun/feed-master/issues/89

This is my first real Go project so ready for comments.
Also I'm not sure which counter in `allStats` I should increment in this case so incremented `ignored`.

### CODE REVIEW

File: app/youtube/service_test.go [hunk 1]
```diff
@@ -111,6 +111,79 @@ func TestService_Do(t *testing.T) {
...
+
+	tmpfile := filepath.Join(os.TempDir(), "test.db")
+	defer os.Remove(tmpfile)
+
+	db, err := bolt.Open(tmpfile, 0o600, &bolt.Options{Timeout: 1 * time.Second})
+	require.NoError(t, err)
+	boltStore := &store.BoltDB{DB: db}
+	svc := Service{
+		Feeds: []FeedInfo{
+			{ID: "channel1", Name: "name1", Type: ytfeed.FTChannel, Filter: FeedFilter{Include: "Prefix2", Exclude: "title3"}},
```
Comment (umputun) [inline app/youtube/service_test.go:146]: I'd suggest adding a case with some regex filters. Maybe for channel2 in addition to the current one

File: app/youtube/service.go [hunk 2]
```diff
@@ -346,6 +363,30 @@ func (s *Service) isNew(entry ytfeed.Entry, fi FeedInfo) (ok bool, err error) {
 	return true, nil
 }
 
+// isRelevant checks if entry matches all filters for the channel feed
+func (s *Service) isRelevant(entry ytfeed.Entry, fi FeedInfo) (ok bool, err error) {
```
Comment (umputun) [inline app/youtube/service.go:367]: I'm not sure the name of the function communicating the intent well. This one is not about relevance, but rather filtering, i.e. isFiltered or isAllowed or smth like this

File: README.md [hunk 3]
```diff
@@ -55,10 +55,11 @@ youtube: # youtube configuration, optional
   rss_location: ./var/rss # location for generated youtube channel's RSS
   channels: # list of youtube channels to download and process
       # id: channel or playlist id, name: channel or playlist name, type: "channel" or "playlist", 
-      # lang: language of the channel, keep: override default keep value 
+      # lang: language of the channel, keep: override default keep value
+      # filter: criteria to include and exclude videos
```
Comment (umputun) [inline README.md:59]: i would all "can be regex" part

**Review by umputun — CHANGES_REQUESTED** [review]
looking good. A few minor things

**Comment by romangr** [comment]
Pushed fixes for all comments

**Review by umputun — APPROVED** [review]
LGTM, thx!

**Pull request merged**